dvc/tree/webdav.py

```diff
@@ -192,6 +192,23 @@ def move(self, from_info, to_info, mode=None):
         # Webdav client move
         self._client.move(from_info.path, to_info.path)
 
+    def upload_fobj(self, fobj, to_info, no_progress_bar=False, **pbar_args):
+        # this is not exactly a chunked upload, since it doesn't quite work
+        # in the way. internally it uses requests, so in theory we should be
+        # able to pass a generator that yields chunks though it is not working
+        # for some reason.
```

Comment:
The comment explains it up. When I checked the source code, I noticed it just passes the `buff=` directly as the `data=` argument of `requests.<method>`, which can actually accept a generator that yields chunks though it didn't work for some reason (~stuck in the test for infinite time), so this doesn't use a chunked approach.

Reply:
btw, if it's an easy fix we have a full control of PyDrive2, we maintain it

Reply:
this is for `webdav`

Reply:
@isidentical Yeah, webdav library is pretty odd with `buff`, even on dowload_to it writes the whole file into the buffer, which seems to defeat the purpose of it. I ran into it too in https://github.com/iterative/dvc/pull/5324 . Looks like we will need to use iter_chunks directly and have some fobj-like object that will read from it on demand.

Reply:
> stuck in the test for infinite time

We use `WSGIServer` for the tests which does not support chunked encoding.

The chunking upload does work if you try it out on a real server. We had support for chunking upload before on `_upload` but we removed it, as the Webdav uses `PUT` to transfer, which does not have good support for chunking upload in older versions (see: https://github.com/iterative/dvc/issues/4796).

Reply:
Thanks for the info @skshetry

Reply:
I have adjusted my comment with this information